File: teams-ai-kit/node/samples/echobot/bot.ts
/**
 * Sample 1 — echobot: a basic AI chat experience.
 *
 * Runs with NO configuration (offline MockModel that echoes smartly), and
 * automatically upgrades to real OpenAI when OPENAI_API_KEY is set.
 *
 *   npm install && npx tsx samples/echobot/bot.ts
 *   curl -s localhost:3978/health
 *   npx tsx samples/client.ts "hello there"
 */

import express from "express";
import { App, AI, MockModel, OpenAIModel, Localization, OpenAIModerator } from "../../src";
import { startDevServer } from "../dev_server";

// --- localization ------------------------------------------------------- //
const locales = new Localization().add("en", {
  welcome: "Hello! I'm EchoBot. Ask me anything, or type /help.",
  blocked: "That message was flagged by moderation and I won't respond to it.",
}).add("es", {
  welcome: "¡Hola! Soy EchoBot. Pregúntame lo que quieras o escribe /help.",
  blocked: "Ese mensaje fue marcado por moderación y no responderé.",
});

// --- model: real when configured, mock otherwise ------------------------ //
const model = process.env.OPENAI_API_KEY
  ? new OpenAIModel({ model: process.env.OPENAI_MODEL })
  : new MockModel({
      say: (text) => `Echo: ${text}`,
    });

// --- the app ------------------------------------------------------------- //
const app = new App();

const ai = new AI({
  model,
  prompt: "You are EchoBot, a friendly assistant in Microsoft Teams. Be concise.",
  moderator: new OpenAIModerator(), // active only when a key is configured
  onBlocked: async (ctx, state) => {
    await ctx.sendActivity(locales.t(state.temp.locale, "blocked"));
  },
});
app.useAI(ai);

app.message("/help", async (ctx, state) => {
  await ctx.sendActivity("Commands:\n/help — this message\n/hi — say hello\nanything else — AI reply (Mock echo offline, OpenAI when configured)");
});

app.message("/hi", async (ctx, state) => {
  await ctx.sendActivity(locales.t(state.temp.locale, "welcome"));
});

// First contact: greet with a card (Teams conversationUpdate).
app.onFallback(async (ctx) => {
  await ctx.sendActivity(locales.t(ctx.activity.locale, "welcome"));
});

// --- web server ----------------------------------------------------------- //
startDevServer(app, Number(process.env.PORT ?? 3978), "echobot", model.constructor.name);

File: teams-ai-kit/python/samples/echobot/bot.py
"""Sample 1 — echobot (Python): a basic AI chat experience.

Runs offline with MockModel; set OPENAI_API_KEY to use real OpenAI.

    pip install -e ".[dev]" && python -m samples.echobot.bot
    python -m samples.client "hello"
"""

import os

from teams_ai_kit import AI, App, Localization, MockModel, OpenAIModerator, OpenAIModel
from samples.dev_server import run_dev_server

locales = (
    Localization()
    .add("en", {"welcome": "Hello! I'm EchoBot (Python). Ask me anything, or type /help.",
                "blocked": "That message was flagged by moderation and I won't respond to it."})
    .add("es", {"welcome": "¡Hola! Soy EchoBot (Python). Pregúntame lo que quieras o escribe /help.",
                "blocked": "Ese mensaje fue marcado por moderación y no responderé."})
)

model = (
    OpenAIModel(model=os.getenv("OPENAI_MODEL"))
    if os.getenv("OPENAI_API_KEY")
    else MockModel(say=lambda text: f"Echo: {text}")
)

app = App()


async def blocked(ctx, state):
    await ctx.send_activity(locales.t(state.temp.get("locale"), "blocked"))


ai = AI(
    model,
    prompt="You are EchoBot, a friendly assistant in Microsoft Teams. Be concise.",
    moderator=OpenAIModerator(),
    on_blocked=blocked,
)
app.use_ai(ai)


@app.message("/help")
async def help_command(ctx, state):
    await ctx.send_activity("Commands:\n/help — this message\n/hi — say hello\nanything else — AI reply (Mock echo offline, OpenAI when configured)")


@app.message("/hi")
async def hi(ctx, state):
    await ctx.send_activity(locales.t(state.temp.get("locale"), "welcome"))


@app.on_fallback
async def fallback(ctx, state):
    await ctx.send_activity(locales.t(state.temp.get("locale"), "welcome"))


if __name__ == "__main__":
    run_dev_server(app, "echobot", model.__class__.__name__, int(os.getenv("PORT", "3978")))
